src/main/webapp/app/overview/exercise-details/course-exercise-details.component.html

```diff
@@ -6,7 +6,9 @@
     <!-- HEADER INFORMATION END -->
     <!-- EXERCISE ACTIONS START -->
     <div class="tab-bar pl-2 pr-2 mb-2 justify-content-between">
-        <jhi-exercise-details-student-actions [courseId]="courseId" [exercise]="exercise" [showResult]="showResults && currentResult"> </jhi-exercise-details-student-actions>
+        <jhi-programming-exercise-student-ide-actions *ngIf="isIntelliJ" [courseId]="courseId" [exercise]="exercise"> </jhi-programming-exercise-student-ide-actions>
```

Comment:
I think this `*ngIf` would be a great use case for an [Angular Directive](https://angular.io/guide/attribute-directives).
([`NgIf` is itself a directive](https://angular.io/api/common/NgIf))


This way, you do not need to do the check and import in every component:
```ts
import { isIntelliJ } from 'app/intellij/intellij';
// ...
readonly isIntelliJ = isIntelliJ;
```

Reply:
I created a jhiIdeFilter directive